### PR TITLE
AIE reset support. Slot separation for PL and AIE only XCLBIN

### PR DESCRIPTION
**Pull request by saifuddin-xilinx**
#### Problem solved by the commit
Whenever there is an issue in AIE array , there is no easy way to reset/reload the AIE array to get AIE Array back to initial state. Customers need  a reliable way to reset and reload AIE array.
For Reset AIE we need PL Only XCLBIN and AIE Only XCLBIN. Both should loaded into different slot. 

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
-- New feature

#### How problem was solved, alternative solutions (if any) and why they were rejected
-- AIE Reset support

#### Risks (if any) associated the changes in the commit
-N/A
#### What has been tested and how, request additional testing if necessary
Basic validation test and specific test case
 
#### Documentation impact (if any)
- N/A

### CODE REVIEW

**Comment by saifuddin-xilinx** [comment]
retest this please

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 1]
```diff
@@ -170,7 +170,7 @@ static int zocl_pr_slot_init(struct drm_zocl_dev *zdev,
 
 		zocl_slot->partial_overlay_id = -1;
 		zocl_slot->slot_idx = i;
-		zocl_slot->slot_type = DOMAIN_PL;
+		zocl_slot->slot_type = ZOCL_PL_ONLY_XCLBIN;
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c:173]: DOMAIN_PL and ZOCL_PL_ONLY_XCLBINs are differrent. I didnt understand how u r mapping this.

File: src/runtime_src/core/edge/drm/zocl/zocl_drv.c [hunk 2]
```diff
@@ -189,7 +189,7 @@ static int zocl_pr_slot_init(struct drm_zocl_dev *zdev,
 		mutex_init(&zocl_slot->slot_xclbin_lock);
 
 		zocl_slot->slot_idx = i;
-		zocl_slot->slot_type = DOMAIN_PS;
+		zocl_slot->slot_type = ZOCL_PS_ONLY_XCLBIN;
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/edge/drm/zocl/zocl_drv.c:192]: Ditto.

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c [hunk 3]
```diff
@@ -702,7 +702,7 @@ zocl_get_slot(struct drm_zocl_dev *zdev, uuid_t *id)
 	struct drm_zocl_slot *zocl_slot = NULL;
 	int i = 0;
 
-	for (i = 0; i < zdev->num_pr_slot; i++) {
+	for (i = 0; i < MAX_PR_SLOT_NUM; i++) {
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c:705]: why to iterate over MAX_PR_SLOT_NUM?

File: src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c [hunk 4]
```diff
@@ -1166,7 +1174,21 @@ zocl_load_sect(struct drm_zocl_dev *zdev, struct axlf *axlf, char __user *xclbin
 static bool
 is_aie_only(struct axlf *axlf)
 {
-	return (axlf->m_header.m_actionMask & AM_LOAD_AIE);
+	if (!xrt_xclbin_get_section_num(axlf, IP_LAYOUT) &&
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c:1177]: have the old check also, axlf->m_header.m_actionMask & AM_LOAD_AIE OR new condition

**Review by chvamshi-xilinx — CHANGES_REQUESTED** [review]

**Comment by chvamshi-xilinx** [comment]
retest this please

**Comment by chvamshi-xilinx** [comment]
Hi @saifuddin-xilinx , Adding "do not review" tag. Please fix all the issues which are discussed offline.

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good

**Comment by saifuddin-xilinx** [comment]
retest this please

**Review by mamin506 — APPROVED** [review]

**Pull request merged**